### PR TITLE
Spanner: Make rows, consume_all and consume_next private

### PR DESCRIPTION
**Pull request by chemelnucfin**
See #4414 and #4204

### CODE REVIEW

**Comment by chemelnucfin** [comment]
@tseaver PTAL, thanks.

File: spanner/google/cloud/spanner_v1/streamed.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ class StreamedResultSet(object):
     """
     def __init__(self, response_iterator, source=None):
         self._response_iterator = response_iterator
-        self._rows = []             # Fully-processed rows
+        self._processed_rows = []             # Fully-processed rows
```
Comment (tseaver) [inline spanner/google/cloud/spanner_v1/streamed.py]: Indent the comment consistently.

File: spanner/tests/system/test_system.py [hunk 2]
```diff
@@ -824,11 +824,11 @@ def test_read_w_manual_consume(self):
         retrieved = 0
         while True:
             try:
-                streamed.consume_next()
+                streamed._consume_next()
```
Comment (tseaver) [inline spanner/tests/system/test_system.py]: I'd argue that we don't want to exercise private methods in a system test.  This test should either be deleted, or else updated to use the expected "public" pattern.

File: spanner/tests/system/test_system.py [hunk 3]
```diff
@@ -946,11 +946,11 @@ def test_execute_sql_w_manual_consume(self):
...
             except StopIteration:
                 break
-            retrieved += len(streamed.rows)
-            streamed.rows[:] = ()
+            retrieved += len(streamed._rows)
+            streamed._rows[:] = ()
```
Comment (tseaver) [inline spanner/tests/system/test_system.py]: Same concern here as with the `read` test.

File: spanner/google/cloud/spanner_v1/streamed.py [hunk 4]
```diff
@@ -48,13 +48,13 @@ def __init__(self, response_iterator, source=None):
         self._source = source       # Source snapshot
 
     @property
-    def rows(self):
+    def _rows(self):
```
Comment (vkedia) [inline spanner/google/cloud/spanner_v1/streamed.py]: why is this method needed? Can we change everywhere this is referenced to directly reference `_processed_rows`

File: spanner/google/cloud/spanner_v1/streamed.py [hunk 5]
```diff
@@ -142,19 +142,19 @@ def consume_next(self):
 
         self._merge_values(values)
 
-    def consume_all(self):
+    def _consume_all(self):
```
Comment (vkedia) [inline spanner/google/cloud/spanner_v1/streamed.py]: why not remove this method? It does not seem to be used anywhere (except tests)

**Comment by chemelnucfin** [comment]
@vkedia @tseaver PTAL, thanks.

File: spanner/tests/unit/test_snapshot.py [hunk 6]
```diff
@@ -252,9 +252,9 @@ def _read_helper(self, multi_use, first=True, count=0):
         else:
             self.assertIsNone(result_set._source)
 
-        result_set._consume_all()
+#        result_set._consume_all()
```
Comment (tseaver) [inline spanner/tests/unit/test_snapshot.py]: Please don't leave commented-out code checked in.

File: spanner/tests/unit/test_snapshot.py [hunk 7]
```diff
@@ -252,9 +252,9 @@ def _read_helper(self, multi_use, first=True, count=0):
...
 
-        self.assertEqual(list(result_set._rows), VALUES)
+        self.assertEqual(list(result_set), VALUES)
```
Comment (tseaver) [inline spanner/tests/unit/test_snapshot.py]: That looks correct, thanks!

File: spanner/tests/unit/test_snapshot.py (hunk 6 above)
Comment (chemelnucfin) [reply spanner/tests/unit/test_snapshot.py]: oops.  fixed.

**Comment by chemelnucfin** [comment]
I just changed the _rows to list.  I will put that in another PR.

**Pull request merged**